src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -0,0 +1,319 @@
+package frc.robot.subsystems;
+
+import com.kauailabs.navx.frc.AHRS;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel;
+import com.revrobotics.RelativeEncoder;
+import com.spikes2212.command.drivetrains.smartmotorcontrollerdrivetrain.SparkMaxTankDrivetrain;
+import com.spikes2212.control.FeedForwardSettings;
+import com.spikes2212.control.PIDSettings;
+import com.spikes2212.control.TrapezoidProfileSettings;
+import com.spikes2212.dashboard.Namespace;
+import com.spikes2212.util.UnifiedControlMode;
+import edu.wpi.first.math.controller.RamseteController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
+import edu.wpi.first.math.kinematics.DifferentialDriveOdometry;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.SerialPort;
+import edu.wpi.first.wpilibj.smartdashboard.Field2d;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import frc.robot.RobotMap;
+
+import java.util.function.Supplier;
+
+public class Drivetrain extends SparkMaxTankDrivetrain {
+
+    public static final double TURN_KS_VOLTAGE = 3.1;
+
+    private static final double WHEEL_DIAMETER_IN_INCHES = 6;
+    private static final double GEAR_RATIO = 1 / 11.16;
+    private static final double INCHES_TO_METERS = 0.0254;
+    private static final double DISTANCE_PER_ROTATION =
+            WHEEL_DIAMETER_IN_INCHES * GEAR_RATIO * Math.PI * INCHES_TO_METERS;
+
+    private static final double TRACK_WIDTH = 0.57;
+
+    /**
+     * Maximum current for the motors. It solved a problem, don't ask me how.
```

Comment:
It solved our problem by stopping the high current spikes (jumps of high current) that made the robot jitter

Reply:
well yes but the maximum current the spark max can physically get is 40